### DISCUSSION TITLE
Tri surf to hex vol mesh

### DISCUSSION THREAD

**Discussion by Sumientra** [SCIInstitute/SCIRun#2261] [Q&A] [answered]
I have a triangular surface mesh that needs to be turned into a mesh consisting of 5 mm cubes. SCIRun only has ConvertQuadSurfToTriSurf and I need the opposite, which is obviously quite a different problem. Does anyone have a good method to do this?

**Comment by jessdtate** [comment, marked as answer]
I think I'm a little confused with the question.  Do you want a mesh of cubes, or quads?  There is a way to get a quad mesh from a trisurf mesh, but it isn't very straightforward and would not guarantee quality elements.  Incidentally, it requires making a cube mesh first.  

Mapping the trisurf onto the elements of a latvol and clip out exterior regions yields a cube mesh.  A few modules can achieve this, including CalculateSignedDistancetoField, CalculateInsideWhichField, or by converting it to a volume with interfacewithtetgen and mapping the value onto the latvol (mapfielddataontoelems, mapfielddatafromsourcetodestination).  Getting the surface (getfieldboundary, getdomainboundary, extractIsosurface) will yield a quad surf mesh, which can be smoothed using fairmesh.  

To guarantee that the boundary is the same, you can use projectPointsOntoMesh to project the meshpoints onto the original trisurf.

**Reply by jessdtate** [reply]
Also, Meshlab should be able to do this, if you want to try that.

**Reply by Sumientra** [reply]
I need a hex vol in the end, but if I had a quad surf that would be easy, so was looking for a solution to either option. Your first solution sounds like it should work and I should have thought of this myself :) Thanks Jess.
